# Human DLPFC {#human_DLPFC}

```{r unref-setup, echo=FALSE, results="asis"}
library(rebook)
chapterPreamble(cache = TRUE)
```


This workflow is a condensed version of the analyses shown in the main analysis chapters, using the same dataset. The workflow uses the most "standard" methods for each step, and contains minimal documentation, to make it easier to see how the steps fit together into a complete analysis pipeline. For more details on any of the individual analysis steps, see the chapters in [Analyis pipeline](#analysis_pipeline).



## Description of dataset {#human_DLPFC_dataset}

This is a 10x Genomics Visium dataset generated from healthy human brain samples from the dorsolateral prefrontal cortex (DLPFC) region.

In the full dataset, there are 12 samples in total, from 3 individuals, with 2 pairs of "spatially adjacent" replicates (serial sections) per individual (i.e. 4 samples per individual). The individuals and spatially adjacent replicates can be considered as blocking factors. Each sample spans the six layers of the cortex plus white matter, in a perpendicular tissue section.

For the examples in this workflow and the analysis chapters, we use only one sample (sample 151673), i.e. a single Visium tissue capture area. This is mainly to reduce computational requirements for compiling the examples.

This dataset is described in our publication @Maynard2020-ke, currently available from [bioRxiv](https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1). The full dataset is publicly available through the [spatialLIBD](http://bioconductor.org/packages/spatialLIBD) Bioconductor package, and the analysis code is provided in the [HumanPilot](https://github.com/LieberInstitute/HumanPilot) GitHub repository. The dataset can also be explored interactively through a [Shiny web app](http://spatial.libd.org/spatialLIBD/).

The subset of data used in this workflow (sample 151673) is also available from the [STdata](https://github.com/lmweber/STdata) package.

For more details on the dataset, including an illustration of the experimental design, see our [paper](https://www.biorxiv.org/content/10.1101/2020.02.28.969931v1) [@Maynard2020-ke] or the [spatialLIBD vignette](http://bioconductor.org/packages/release/data/experiment/vignettes/spatialLIBD/inst/doc/spatialLIBD.html).



## Load data

Load the dataset from the [STdata](https://github.com/lmweber/STdata) package. Note that this is a subset containing only one sample (sample 151673) from the full dataset.

[TO DO: convert `STdata` package to an `ExperimentHub package; currently is a temporary wrapper around Drobox links]

[TO DO: convert to using `SpatialExperiment` class instead of `SingleCellExperiment`]

```{r load_data, message=FALSE}
library(ExperimentHub)
library(STdata)

ehub <- ExperimentHub()

spe <- load_data("human_DLPFC")
spe
```



## Quality control (QC)

Calculate spot-level QC metrics using `scater` [@McCarthy2017-zd]. See [Quality control](#quality_control) for more details, including explanations of the QC metrics.

```{r QC_calculate, message=FALSE}
library(scater)

# identify mitochondrial genes
is_mito <- grepl("^MT-", rowData(spe)$gene_name)
table(is_mito)

# calculate per-cell QC metrics (treating spots as equivalent to cells)
df <- perCellQCMetrics(spe, subsets = list(mito = is_mito))
head(df, 3)

# store per-cell QC metrics in colData of SCE object
spe <- addPerCellQC(spe, subsets = list(mito = is_mito))
```

Select thresholds for QC metrics. For more details, including how to use exploratory visualizations to select the thresholds, see [Quality control](#quality_control). Note that we use relatively relaxed thresholds, since the exploratory visualizations showed that more stringent thresholds tended to remove groups of spots correlated with biologically meaningful spatial patterns.

```{r QC_thresholds}
# select QC thresholds
qc_lib_size <- colData(spe)$sum < 500
qc_detected <- colData(spe)$detected < 250
qc_mito <- colData(spe)$subsets_mito_percent > 30
qc_cell_count <- colData(spe)$cell_count > 12

# number of discarded spots for each metric
apply(cbind(qc_lib_size, qc_detected, qc_mito, qc_cell_count), 2, sum)

# combined set of discarded spots
discard <- qc_lib_size | qc_detected | qc_mito | qc_cell_count
table(discard)

# store in object
colData(spe)$discard <- discard
```

Plot the set of discarded spots in the original x-y coordinates, to confirm that the spatial distribution of the discarded spots does not correspond to any obvious biological patterns in the data, which would indicate that we are removing biological signal.

```{r QC_check, fig.height=4, message=FALSE}
library(spatzli)

# check spatial pattern of combined set of discarded spots
plotQCspots(spe, discard = "discard")
```

Remove the low-quality spots from the object.

```{r QC_remove}
# remove combined set of low-quality spots
spe <- spe[, !colData(spe)$discard]
dim(spe)
```



## Normalization

Calculate log-transformed normalized counts, using pool-based size factors and deconvolution to spot level. We use normalization methods for scRNA-seq from `scater` [@McCarthy2017-zd] and `scran` [@Lun2016-dn], treating spots as equivalent to cells. Note that since we have only a single sample, there are no blocking factors in the experimental design. For more details, see [Normalization](#normalization).

```{r normalization, fig.height=4, message=FALSE}
library(scran)

# clustering for pool-based size factors
set.seed(123)
clusters <- quickCluster(spe)
table(clusters)

# calculate size factors and store in SCE object
spe <- computeSumFactors(spe, cluster = clusters)

summary(sizeFactors(spe))
hist(sizeFactors(spe), breaks = 20)

# calculate logcounts (log-transformed normalized counts) and store in SCE object
spe <- logNormCounts(spe)

assayNames(spe)
```



## Feature selection

Select top highly variable genes (HVGs). We use methods from `scran` [@Lun2016-dn], treating spots as equivalent to cells, and considering only molecular features (gene expression). Newer methods adapted for spatial data will be described in [Feature selection](#feature_selection). We also first filter mitochondrial genes, since these are very highly expressed and not of biological interest in this dataset.

```{r feature_selection_mito}
# remove mitochondrial genes
spe <- spe[!is_mito, ]
dim(spe)
```

```{r feature_selection_HVGs, fig.height=5, message=FALSE}
# fit mean-variance relationship
dec <- modelGeneVar(spe)

# visualize mean-variance relationship
fit <- metadata(dec)
plot(fit$mean, fit$var, 
     xlab = "mean of log-expression", ylab = "variance of log-expression")
curve(fit$trend(x), col = "dodgerblue", add = TRUE, lwd = 2)

# select top HVGs
top_hvgs <- getTopHVGs(dec, prop = 0.1)
length(top_hvgs)
```



## Dimensionality reduction

Apply principal component analysis (PCA) to the set of top HVGs to reduce the dimensionality of the dataset, and retain the top 50 principal components (PCs) for further downstream analyses. This is done both to reduce noise and to improve computational efficiency during downstream analyses. We also calculate UMAP on the set of top 50 PCs and retain the top 2 UMAP components for visualization purposes.

We use the computationally efficient implementation of PCA available in `scater` [@McCarthy2017-zd], which uses randomization and therefore requires a random seed for reproducibility.

```{r dimensionality_reduction_PCA}
# calculate PCA
set.seed(123)
spe <- runPCA(spe, subset_row = top_hvgs)

reducedDimNames(spe)
dim(reducedDim(spe, "PCA"))
```

```{r dimensionality_reduction_UMAP}
# calculate UMAP on top 50 PCs
set.seed(123)
spe <- runUMAP(spe, dimred = "PCA")

reducedDimNames(spe)
dim(reducedDim(spe, "UMAP"))

# update column names (for easier plotting)
colnames(reducedDim(spe, "UMAP")) <- paste0("UMAP", 1:2)
```



## Clustering

Next, we perform clustering to identify cell types. Here, we use molecular features only (i.e. the top 50 PCs calculated on the set of top HVGs). Alternative methods that also take into account spatial features are described in [Clustering](#clustering). We use the graph-based clustering implementation and Walktrap method provided in `scran` [@Lun2016-dn].

From the visualizations, we can see that the clustering reproduces the known biological structure (cortical layers), although not perfectly. The clusters are also separated in UMAP space.

```{r clustering_calculate}
# graph-based clustering
set.seed(123)
k <- 10
g <- buildSNNGraph(spe, k = k, use.dimred = "PCA")
g_walk <- igraph::cluster_walktrap(g)
clus <- g_walk$membership
table(clus)

# store cluster labels in object (in column "label")
colLabels(spe) <- factor(clus)
```

```{r clustering_plots, fig.height=4.5, fig.width=5.5}
# plot clustering in x-y space
plotClustersSpots(spe, cluster_id = "label")

# plot ground truth layers in x-y space
plotClustersSpots(spe, cluster_id = "ground_truth")

# plot clustering in UMAP space
plotClustersDimRed(spe, type = "UMAP", cluster_id = "label")
```



## Interpretation

Identify marker genes by testing genes for differential expression between clusters. We use the `findMarkers` implementation in `scran` [@Lun2016-dn], using a binomial test, which tests for genes that differ in the proportion of expressing genes (expressed vs. not expressed) between clusters. This is a more stringent test than the default t-tests, and tends to select genes that are easier to interpret and validate experimentally.

```{r marker_genes}
# set gene names for plotting
rownames(spe) <- rowData(spe)$gene_name

# test for marker genes
markers <- findMarkers(spe, test = "binom", direction = "up")

# returns a list with one DataFrame per cluster
markers
```

```{r marker_genes_heatmap, fig.width=5, message=FALSE}
library(pheatmap)

# plot log-fold changes for one cluster over all other clusters
interesting <- markers[[1]]
best_set <- interesting[interesting$Top <= 5, ]
logFCs <- getMarkerEffects(best_set)

pheatmap(logFCs, breaks = seq(-5, 5, length.out = 101))
```

```{r marker_genes_expression, fig.width=7}
# plot log-transformed normalized expression of top genes for one cluster
top_genes <- head(rownames(interesting))

plotExpression(spe, x = "label", features = top_genes)
```


