# Spatial transcriptomics {#spatial_transcriptomics}

```{r setup, echo=FALSE, results="asis"}
library(rebook)
chapterPreamble()
```


Spatial transcriptomics represents a new class of technologies that allows gene expression to be measured across the complete transcriptome with spatial resolution. This facilitates new types of experiments to investigate spatial patterns of gene expression in an exploratory manner.


## Technologies

There are several technologies that fall under the umbrella of spatial transcriptomics. This section describes some key features of the most popular technologies. For the analysis pipeline and workflows in the main parts of this book, we use data generated with the 10x Genomics Visium platform.



### 10x Genomics Visium

The [Visium](https://www.10xgenomics.com/spatial-transcriptomics/) platform was commercially released by [10x Genomics](https://www.10xgenomics.com/) in late 2019. Along with the platform itself, 10x Genomics has released analysis software such as [Space Ranger](https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/what-is-space-ranger), and a number of other resources such as [example datasets](https://www.10xgenomics.com/resources/datasets/).

The Visium platform measures gene expression at a two-dimensional grid of "spots" located on a tissue slide. Fresh-frozen tissue is placed onto the slide, fixed, stained, and permeabilized. Each spot contains millions of spatially-barcoded capture oligonucleotides, which bind to mRNAs from the tissue. A cDNA library is then generated for sequencing, which includes the spatial barcodes, allowing reads to be mapped back to their spatial location.

The size of the tissue slide is approximately 6.5mm by 6.5mm, and contains around 5000 barcoded spots. Spots are located in a regular hexagonal arrangement, with each spot 55µm in diameter, and 100µm center to center. This gives a resolution roughly on the order of single cells, depending on the tissue type. Each slide contains 4 capture areas (6.5mm x 6.5mm each). The following figure provides an illustration.


```{r Visium-screenshot, fig.wide=TRUE, echo=FALSE, fig.cap="Illustration of 10x Genomics Visium slide. Image source: 10x Genomics Visium technology brochure, available from: https://www.10xgenomics.com/spatial-transcriptomics/"}
library(rebook)
knitr::include_graphics(
  file.path("https://raw.githubusercontent.com/lmweber/OSTA-release/master/images", 
            "Visium_screenshot.png")
)
```


More details are available on the [10x Genomics Visium](https://www.10xgenomics.com/spatial-transcriptomics/) commercial website.

The Visium platform extends a previous implementation of this technology, which was simply referred to as "Spatial Transcriptomics", and published by @Stahl2016-sk.

Due to the commercial availability, high resolution, transcriptome-wide measurement, and [relatively simple experimental procedures compared to other technologies](https://www.10xgenomics.com/spatial-transcriptomics/), we expect the Visium platform to become widely used in the coming years, and therefore focus our main analysis pipeline chapters and workflows on Visium datasets.


### Slide-seqV2

Paragraph on Slide-seqV2 and Slide-seq

Note much lower number of genes per spot than Visium: [Valentine Svensson blog post](http://www.nxn.se/valent/2020/5/11/comparing-slide-seq-and-slide-seqv2-counts)


### MERFISH and seqFISH+

Paragraph on MERFISH (latest iteration) and seqFISH+

Allows subcellular resolution


### Other spatial gene expression technologies

Brief mention of other spatial gene expression technologies, e.g. RNAscope (for completeness)


### Other spatial omics technologies

Brief mention of other spatial omics technologies, e.g. imaging mass cytometry



## Data format and objects

As discussed in the [Introduction](#singlecellexperiment), we use the `SingleCellExperiment` Bioconductor object class to store our Visium datasets for the main analysis pipeline and workflows. The datasets are described in the next chapter.

However, this relies on a number of simplifying assumptions, which are partly specific to the Visium platform. This may be revised in the future if more specialized Bioconductor object classes are developed for spatial transcriptomics data.

Currently, we store Visium data in the `SingleCellExperiment` class as follows:

- "spots" are stored in columns (instead of cells as in scRNA-seq; note that spots may contain zero, one, or more actual cells)
- image coordinates corresponding to each spot are stored in `colData`
- genes are stored in rows (same as in scRNA-seq)
- `counts` and `logcounts` are stored in `assays` (same as in scRNA-seq)
- histology images may optionally be stored in `metadata` (but may be very large)

Note that the resolution of Visium data is roughly on the order of single cells, and the same set of genes is measured at each spot. It is not yet clear how we would handle subcellular resolution data, or how to store data where different genes are measured at different spatial coordinates (e.g. fluorescent imaging at subcellular resolution).


