# Acknowledgments {#sec-acknowledgments}


## Contributors


### 'Best Practices for Spatial Transcriptomics Analysis with Bioconductor' online book

- [Lukas M. Weber](https://lmweber.org/), *Johns Hopkins Bloomberg School of Public Health, Baltimore, MD, USA*
- [Leonardo Collado-Torres](https://lcolladotor.github.io/), *Lieber Institute for Brain Development, Baltimore, MD, USA*
- [Stephanie C. Hicks](https://www.stephaniehicks.com/), *Johns Hopkins Bloomberg School of Public Health, Baltimore, MD, USA*

As well as additional [GitHub contributors](https://github.com/lmweber/BestPracticesST/graphs/contributors). Thank you for your contributions!


### SpatialExperiment data class

- Dario Righelli, *Department of Statistical Sciences, University of Padova, Padova, Italy*
- Helena L. Crowell, *Department of Molecular Life Sciences, University of Zurich, Zurich, Switzerland*
- Aaron Lun, *Genentech, South San Francisco, CA, USA*
- Stephanie C. Hicks, *Johns Hopkins Bloomberg School of Public Health, Baltimore, MD, USA*
- Davide Risso, *Department of Statistical Sciences, University of Padova, Padova, Italy*



## Additional acknowledgments

- The 'Best Practices for Spatial Transcriptomics Analysis with Bioconductor' online book is inspired by the [Orchestrating Single-Cell Analysis with Bioconductor (OSCA)](https://bioconductor.org/books/release/OSCA/) online book, which provides a detailed overview and examples of analysis workflows for single-cell RNA sequencing data using R and Bioconductor.

- We are especially grateful to the [Bioconductor](https://bioconductor.org/) community and the [Bioconductor Core Team](https://www.bioconductor.org/about/core-team/).
