# Spatial transcriptomics {#sec-spatial-transcriptomics}


## Overview

Spatial transcriptomics (or spatially-resolved transcriptomics) refers to a set of recently developed high-throughput genomic technologies, which enable the measurement of up to transcriptome-scale gene expression at spatial resolution -- i.e. measuring the expression of up to thousands of genes at up to thousands of measurement locations, usually on two-dimensional tissue sections.

A number of technological platforms have now been developed, which vary widely in terms of the technologies used and experimental procedures, as well as the number of genes that can be measured and the spatial resolution, and sensitivity. Spatially-resolved transcriptomics was named the [Method of the Year 2020](https://doi.org/10.1038/s41592-020-01042-x) by *Nature Methods*, and has found widespread application in numerous biological systems.

Platforms may be grouped into 'sequencing-based' and 'molecule-based' (or 'imaging-based') platforms. These are described in more detail below.

For recent reviews of available technological platforms, computational analysis methods, and outstanding challenges, see @Bressan2023 or @Moses2022.



## Overview

In this book, we focus on commercially available technological platforms, since these are the most easily accessible and widely used. In this chapter, we provide a short overview and details on several key platforms.

```{r, echo=FALSE, out.width = "100%", fig.align="center", fig.cap="Illustration of potential uses of spatially-resolved transcriptomics data and how it can be combined with single cell/nucleus RNA-seq to study tissues of interest such as the human brain. Source: @Maynard2020a."}
knitr::include_graphics("https://raw.githubusercontent.com/lmweber/BestPracticesST-resources/main/images/Maynard2020_spatial.png")
```



## Sequencing-based platforms

Sequencing-based platforms capture messenger RNA (mRNA) molecules at a large set (up to thousands) of measurement locations on a tissue slide, tag the mRNAs with unique spatial barcodes for each  measurement location, and generate a readout by next-generation sequencing.

These platforms can provide full-transcriptome gene resolution. The spatial resolution varies between platforms, and depends on the size and spacing between the spatial capture locations. Depending on the spatial resolution and the tissue cell density, each spatial measurement location may capture multiple cells.

Spatial measurement locations for sequencing-based platforms are often referred to as 'spots' or 'beads'. In this book, we will often use the terminology 'spots'.


### 10x Genomics Visium

The [10x Genomics Visium](https://www.10xgenomics.com/products/spatial-gene-expression) platform measures transcriptome-wide gene expression at a grid of spatial locations (referred to as spots) on a tissue capture area. Either fresh-frozen or formalin-fixed paraffin-embedded (FFPE) tissue may be used. Each spot contains millions of spatially-barcoded capture oligonucleotides, which bind to mRNAs from the tissue. A cDNA library is then generated for sequencing, which includes the spatial barcodes, allowing reads to be mapped back to their spatial locations.

The array dimensions are 6.5mm x 6.5mm, with around 5000 barcoded spots. Spots are 55µm in diameter and spaced 100µm center-to-center in a hexagonal grid arrangement. The number of cells per spot depends on the tissue cell density, e.g. around 0-10 for human brain or ~50 for mouse brain. Each Visium slide contains four tissue capture areas. The following figure provides an illustration.

Histology images generated from hematoxylin and eosin (H&E) staining can be used to identify anatomical and cell morphological features such as the number of cells per spot.

```{r Visium-schematic, echo=FALSE, fig.wide=TRUE, fig.align="center", fig.cap="Schematic of 10x Genomics Visium platform. Source: [10x Genomics Visium website](https://www.10xgenomics.com/spatial-transcriptomics/)"}
knitr::include_graphics("https://raw.githubusercontent.com/lmweber/BestPracticesST-resources/main/images/Visium.png")
```


### Curio Seeker

The [Curio Seeker](https://curiobioscience.com/product/) platform is a commercial implementation of the Slide-seqV2 @Stickels2021 platform.

This platform captures mRNA molecules at a set of randomly placed measurement locations (referred to as 'beads') on a tissue slide (referred to as a 'tile') at high spatial resolution. The tile dimensions are 3 mm x 3 mm, and the spatial resolution consists of tightly packed 10 μm diameter beads.



## Molecule-based platforms

Molecule-based (or imaging-based) platforms identify the spatial coordinates of individual RNA molecules by combinatorial in situ hybridization (ISH) or in situ sequencing (ISS), for targeted sets of hundreds to thousands of genes at up to sub-cellular spatial resolution.

Image segmentation to identify the boundaries of individual cells and assign RNA molecules to cells represents a key step during analyses. For downstream analyses, gene counts may be aggregated to cellular (single-cell) resolution, or analyses may be performed directly at the level of the individual RNA molecules.

Panel design to select sets of informative genes in a given biological context is an important consideration during experimental design, with several commerical options now available.


### 10x Genomics Xenium

Additional details on the 10x Genomics Xenium platform are available from [10x Genomics](https://www.10xgenomics.com/platforms/xenium).


### Vizgen MERSCOPE

Additional details on the Vizgen MERSCOPE platform are available from [Vizgen](https://vizgen.com/products/).


### NanoString CosMx

Additional details on the NanoString CosMx platform are available from [NanoString](https://nanostring.com/products/cosmx-spatial-molecular-imager/).
